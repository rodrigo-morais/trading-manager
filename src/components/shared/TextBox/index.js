import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'


const Container = styled.div`
  display: inline-block;
  width: 100%;
`

const Label = styled.label`
  font-size: 18px;
  font-weight: bold;
  margin-right: 8px;
`

const Text = styled.input.attrs({
  type: 'text',
})`
  width: 40%;
  color: #000000;
  font-size: 16px;
`

const addColon = (label) => label.slice(-1) === ':' ? label : `${label}:`

const TextBox = ({ className, label, disabled }) => (
  <Container className={className}>
    <Label htmlFor={label}>{addColon(label)}</Label>
    <Text name={label} id={label} disabled={disabled} />
  </Container>
)

TextBox.defaultProps = {
  disabled: false,
}

TextBox.defaultProps = {
  className: '',
}

TextBox.propTypes = {
  className: PropTypes.string,
	label: PropTypes.string.isRequired,
  disabled: PropTypes.bool,
}

export default TextBox
