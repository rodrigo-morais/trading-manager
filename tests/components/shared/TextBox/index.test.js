import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'

import TextBox from '../../../../src/components/shared/TextBox'

describe('TextBox', () => {
  describe('Rendering', () => {
    describe('enabled DropDown', () => {
      const wraper = (
          <TextBox
            label="Ativo"
          />
      )

      test('renders according to design', () => {
        const component = renderer.create(wraper)
        const tree = component.toJSON()

        expect(tree).toMatchSnapshot()
      })

      test('text box enabled', () => {
        const component = mount(wraper)
        expect(component.find('input.text').at(0).disabled).toBeFalsy()
      })

      test('text box label is equal "Ativo"', () => {
        const component = mount(wraper)
        expect(component.find('label').text()).toEqual('Ativo:')
      })
    })

    describe('disabled DropDown', () => {
      const wraper = (
          <TextBox
            label="Ativo"
            disabled
          />
      )

      test('renders according to design', () => {
        const component = renderer.create(wraper)
        const tree = component.toJSON()

        expect(tree).toMatchSnapshot()
      })

      test('drop down enabled', () => {
        const component = mount(wraper)
        expect(component.find('input.text').at(0).disabled).toBeFalsy()
      })

      test('text box label is equal "Ativo"', () => {
        const component = mount(wraper)
        expect(component.find('label').text()).toEqual('Ativo:')
      })
    })
  })
})
