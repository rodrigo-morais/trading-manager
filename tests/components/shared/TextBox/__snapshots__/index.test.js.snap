// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextBox Rendering disabled DropDown renders according to design 1`] = `
<div
  className="sc-bdVaJa lhrPED"
>
  <label
    className="sc-bwzfXH ebnOHd"
    htmlFor="Ativo"
  >
    Ativo:
  </label>
  <input
    className="sc-htpNat iZIyaN"
    disabled={true}
    id="Ativo"
    name="Ativo"
    type="text"
  />
</div>
`;

exports[`TextBox Rendering enabled DropDown renders according to design 1`] = `
<div
  className="sc-bdVaJa lhrPED"
>
  <label
    className="sc-bwzfXH ebnOHd"
    htmlFor="Ativo"
  >
    Ativo:
  </label>
  <input
    className="sc-htpNat iZIyaN"
    id="Ativo"
    name="Ativo"
    type="text"
  />
</div>
`;
